src/application/deployer/index.go

```diff
@@ -12,7 +12,7 @@ import (
 // StartDeploy starts the deployment process
 // nolint gocyclo
 func StartDeploy(deployConfig deploy.Config) error {
-	err := validateConfigs(deployConfig)
+	err := ValidateConfigs(deployConfig)
```

Comment:
Can be dealt with separately, but I think this function should be updated to check the terraform file contents are up to date here (instead of after pushing the docker images). I also think we should not write any terraform updates to file as we will always role them back. Thus we could diff it in the same way we are with docker-compose.

Reply:
As part of that I think the ValidateConfigs function should move into the generate file instead of in here as I don't think we need to validate configs as part of deploy and only as part of the generate process.

Reply:
We aren't rolling back file changes anymore - we got rid of that when we switched to Terraform 0.11.0: https://github.com/Originate/exosphere/commit/3218aee19bf720b1202b6ac82d92f9b187c78558

I also didn't like that feature much anyways. I think it was much more helpful to see those changes written to file that way you could diff them with source control.

Reply:
Nevermind, with checking for terraform updates before running Terraform commands we should be able to preserve the previous file. And we can always diff with source control by running `exo generate terraform`